include/jemalloc/internal/tsd_types.h

```diff
@@ -291,7 +291,8 @@ a_name##tsd_cleanup_wrapper(void) {					\
 	if (wrapper == NULL) {						\
 		return false;						\
 	}								\
-	if (a_cleanup != malloc_tsd_no_cleanup &&			\
+	if (/* a_cleanup != malloc_tsd_no_cleanup && */			\
+	    /* Verify compiler (mingw32+gcc) bug.*/			\
```

Comment:
Yeah the overhead should be minimal -- happen only once per thread and only a dull function call (if it's not optimized away).

In fact, since currently we don't use the _no_cleanup. It doesn't matter at all.